server/xpub-model/entities/file/index.js

```diff
@@ -25,23 +26,33 @@ class File extends BaseModel {
     }
   }
 
-  async save() {
+  async save(userId) {
     await this.$query().upsertGraphAndFetch(this)
 
+    const manuscript = await Manuscript.find(this.manuscriptId, userId)
+
+    if (!manuscript) {
+      throw new Error(
+        `Unable to find Manuscript of ID: ${
+          this.manuscriptId
+        } assigned to File: ${this.id}`,
+      )
+    }
+
+    await manuscript.validate()
```

Comment:
suggestion:
- pass in the `Manuscript` object as an argument here (to make the runtime dependency explicit, if the layering/framework allows it)
- transform this into `manuscript.fileSaved(this.filename, this.status)` (probably needs fewer arguments) which will be the domain event handler

This is still transactional on both objects, but eventually it could make the boundary easier to see when supporting parallel file uploads

Reply:
In fact, we've avoided including models with models except for the AuditLog model, which can get included in other models.

If we want to update models in conjunction with each other, we should do this within `xpub-controller`

Reply:
I'll port all of the uploads / deletes to the `manuscript` controller and call validate at that level

Reply:
Interesting, let's see how this will look like after that